forestci/forestci.py

```diff
@@ -52,28 +62,48 @@ def calc_inbag(n_samples, forest):
     """
 
     if not forest.bootstrap:
-        e_s = "Cannot calculate the inbag from a forest that has "
-        e_s = " bootstrap=False"
+        e_s = "Cannot calculate the inbag from a forest that has bootstrap=False"
         raise ValueError(e_s)
 
-    n_trees = forest.n_estimators
-    inbag = np.zeros((n_samples, n_trees))
-    sample_idx = []
-    n_samples_bootstrap = _get_n_samples_bootstrap(
-        n_samples, forest.max_samples
-    )
-
-    for t_idx in range(n_trees):
-        sample_idx.append(
-            _generate_sample_indices(forest.estimators_[t_idx].random_state,
-                                     n_samples, n_samples_bootstrap))
-        inbag[:, t_idx] = np.bincount(sample_idx[-1], minlength=n_samples)
-    return inbag
-
-
-def _core_computation(X_train, X_test, inbag, pred_centered, n_trees,
-                      memory_constrained=False, memory_limit=None,
-                      test_mode=False):
+    if isinstance(forest, BaseForest):
+        n_trees = forest.n_estimators
+        inbag = np.zeros((n_samples, n_trees))
+        sample_idx = []
+        n_samples_bootstrap = _get_n_samples_bootstrap(n_samples, forest.max_samples)
+
+        for t_idx in range(n_trees):
+            sample_idx.append(
+                _generate_sample_indices(
+                    forest.estimators_[t_idx].random_state,
+                    n_samples,
+                    n_samples_bootstrap,
+                )
+            )
+            inbag[:, t_idx] = np.bincount(sample_idx[-1], minlength=n_samples)
+        return inbag
+    elif isinstance(forest, BaseBagging):
+        samples_and_counts = [
+            np.unique(sample, return_counts=True)
+            for sample in forest.estimators_samples_
+        ]
+
+        dataframes = [
+            pd.DataFrame(index=index, data=data) for index, data in samples_and_counts
+        ]
+
+        return pd.concat(dataframes, axis="columns").fillna(0).astype(np.int).values
```

Comment:
Am I correct that this is the only place that pandas is used? Is it here just to find nan values?

Reply:
You're correct that was the only pandas dependency. I was using it for `fillna` and then also for `concat` to merge on the sample indices. But I just pushed a commit that gets rid of the pandas dependency.